src/queue.rs

```diff
@@ -1260,18 +1276,51 @@ impl InProgressTask {
         .fetch_one(executor)
         .await?)
     }
+
+    pub(crate) async fn record_heartbeat<'a, E>(&self, executor: E) -> Result
+    where
+        E: PgExecutor<'a>,
+    {
+        sqlx::query!(
+            r#"
+            update underway.task
+            set updated_at = now(),
+                last_heartbeat_at = now()
+            where id = $1
```

Comment:
this still confuses me, I'd expect `id` to be the "real" id, eg it is unique

Reply:
It is unique per queue name.

In part this is a semantical quality that's "more correct" but there's also some practical reasons to do it this way. For example, this means the same task (i.e. per ID) can be in multiple queues but that for each queue only one unique copy can exist.

Reply:
> this means the same task (i.e. per ID) can be in multiple queues
This confuses me even more, I would expect id = identity = lifecycle, so it makes it tricky to figure out which job and how updates `task` status...
but lets take this discussion out of the PR, look good to me, btw :)